samples/extensions/layer_settings/layer_settings.cpp

```diff
@@ -0,0 +1,315 @@
+/* Copyright (c) 2025, Holochip Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "layer_settings.h"
+
+#include "common/vk_common.h"
+#include "common/vk_initializers.h"
+
+#include <array>
+
+LayerSettingsSample::~LayerSettingsSample()
+{
+	if (debug_messenger_ != VK_NULL_HANDLE && has_instance())
+	{
+		VkInstance instance  = get_instance().get_handle();
+		auto       fpDestroy = reinterpret_cast<PFN_vkDestroyDebugUtilsMessengerEXT>(
+            vkGetInstanceProcAddr(instance, "vkDestroyDebugUtilsMessengerEXT"));
```

Comment:
Why do you explicitly get the function pointer? As volk is initialized by the framework, just calling
`		vkDestroyDebugUtilsMessengerEXT(instance, debug_messenger_, nullptr);
`works as well